packages/core/src/types.ts

```diff
@@ -897,14 +897,20 @@ export type PropertyAssigner<TContext, TEvent extends EventObject> = {
     | TContext[K];
 };
 
-export type Mapper<TContext, TEvent extends EventObject> = (
+export type Mapper<TContext, TEvent extends EventObject, TParams extends {}> = (
   context: TContext,
   event: TEvent
-) => any;
+) => TParams;
 
-export type PropertyMapper<TContext, TEvent extends EventObject> = Partial<{
-  [key: string]: ((context: TContext, event: TEvent) => any) | any;
-}>;
+export type PropertyMapper<
+  TContext,
+  TEvent extends EventObject,
+  TParams extends {}
```

Comment:
actually not sure how much of a value this gives - in the codebase, this is always supplied as `any`, but in theory this could be used in userland or in the codebase in the future in some specific cases to provide stricter type for a given call